spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -161,25 +188,33 @@ public File getFile() throws IOException {
 
 	@Override
 	public long contentLength() throws IOException {
-		return throwExceptionForNullBlob(getGoogleStorageObject()).getSize();
+		return throwExceptionForNullBlob(getGcsBlob()).getSize();
 	}
 
 	@Override
 	public long lastModified() throws IOException {
-		return throwExceptionForNullBlob(getGoogleStorageObject()).getUpdateTime();
+		return throwExceptionForNullBlob(getGcsBlob()).getUpdateTime();
 	}
 
 	@Override
-	public Resource createRelative(String relativePath) throws IOException {
+	public GoogleStorageResourceObject createRelative(String relativePath) throws IOException {
 		int lastSlashIndex = this.location.lastIndexOf("/");
 		String absolutePath = this.location.substring(0, lastSlashIndex + 1) + relativePath;
-		return new GoogleStorageResourceObject(this.storage, absolutePath, this.createBlobIfNotExists);
+
+		GoogleStorageResourceObject blobResource = new GoogleStorageResourceObject(
+				this.storage, absolutePath, this.createBlobIfNotExists);
+
+		if (!blobResource.exists()) {
+			blobResource.create();
```

Comment:
You're giving `this.createBlobIfNotExists`, which may be false, to the relative blob, but you always create it in this line, so it's like that setting no longer matters. Maybe you should just pass `true` in that constructor on the previous line?

Reply:
`createBlobIfNotExists` determines if an object should be **auto**created, not if it should be created.

In other words, `createBlobIfNotExists` should only be taken into account for the auto-creation in `getOutputStream()`.

When users call `createRelative()`, they're saying they want to create an object, so that's what we do.